### PR TITLE
frotz: add test block

### PR DESCRIPTION
**Pull request by anag004**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

Added a simple 'test do' block for Frotz which runs a small test.z5 file. 
-----

### CODE REVIEW

**Comment by anag004** [comment]
For the purpose of testing the formula, I created an additional file test.z5 in `/usr/local/Cellar/frotz/2.44`. How do I edit the formula to include this file in the standard installation?

**Comment by igas** [comment]
Sorry, I'm not familiar with the tool itself. Is it a plain text file? is it big? You should either write it in the formula itself or use `resource` to download it.

**Comment by igas** [comment]
I can see example files in https://github.com/DavidGriffith/frotz/tree/master/src/test so you can do something like:

https://github.com/Homebrew/homebrew-core/blob/830db7ae79167ab9036d9267980ecd80c3fc4134/Formula/blockhash.rb#L19-L22

and

https://github.com/Homebrew/homebrew-core/blob/830db7ae79167ab9036d9267980ecd80c3fc4134/Formula/blockhash.rb#L31

**Comment by anag004** [comment]
Okay @igas I will do that. I changed the title of the PR as well. Thanks for the warm welcome to Home-brew :)

**Comment by anag004** [comment]
@igas could you explain what the sha256 hash does in the resource block? What is it a hash of?

**Comment by SMillerDev** [comment]
Not igas but the hash is a hash of the resource you're downloading

**Comment by anag004** [comment]
The automated checks indicate that there is a mismatch in the SHA256 checksum. On my local machine, the checksum `c9cf74c3ae0e89c91d323cca9c77e65ae0f4b3de1f6db42f6e208094a8dfdf92` passes `brew test frotz`. In the failed tests, I get the following error messages (in two of them):

In `brew-test-bot.sierra.test frotz`:

`ChecksumMismatchError: SHA256 mismatch
Expected: c9cf74c3ae0e89c91d323cca9c77e65ae0f4b3de1f6db42f6e208094a8dfdf92
Actual: cfc156cc601e86d9637c9e5213c578691cb22adbf96866a9d7682fb23900bf34`

In `brew-test-bot.high_sierra.test frotz`:

`ChecksumMismatchError: SHA256 mismatch
Expected: c9cf74c3ae0e89c91d323cca9c77e65ae0f4b3de1f6db42f6e208094a8dfdf92
Actual: 2c522f44914a7c83cb2d0c0191b5073681d8fea68fa6e389eab31b6f84bd6bf3`

Can I do anything to fix this? Why does the bot report two different actual values of the checksum?

**Comment by Moisan** [comment]
The link of your new resource block points to the github page, which contains a lot of html that could change between two downloads. You probably want the raw `test.z5` file, which can be found at this url:
```
https://github.com/anag004/FrotzTestFile/blob/master/test.z5?raw=true
```

Also you can see on the jenkins log that the failing command wasn't `brew test` but 
```
brew fetch --retry frotz --build-bottle --force
```
 You can test it locally if you want.

**Comment by anag004** [comment]
Thanks @Moisan . I've updated the resource link to download the raw file.

File: Formula/frotz.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ class Frotz < Formula
     sha256 "6506ec53859a859eb71ab33511a11e9c957e0749cb25efa06d63d6c80d000b4a" => :el_capitan
     sha256 "887efd67a050038ffb15df8c713eaa40fcbfcc69d36f58373981dba3ccae422b" => :yosemite
   end
+  resource "testdata" do
```
Comment (javian) [inline Formula/frotz.rb]: You need to add an empty line here

File: Formula/frotz.rb [hunk 2]
```diff
@@ -20,8 +24,11 @@ def install
       s.change_make_var! "CONFIG_DIR", etc
       s.change_make_var! "MAN_PREFIX", share
     end
-
     system "make", "frotz"
     system "make", "install"
   end
+  test do
```
Comment (javian) [inline Formula/frotz.rb]: You need to add an empty line here

File: Formula/frotz.rb [hunk 3]
```diff
@@ -20,8 +24,11 @@ def install
       s.change_make_var! "CONFIG_DIR", etc
       s.change_make_var! "MAN_PREFIX", share
     end
-
```
Comment (javian) [inline Formula/frotz.rb]: You can restore this line

**Review by javian — CHANGES_REQUESTED** [review]

**Comment by anag004** [comment]
@javian Added the lines you requested. Just out of curiosity, you requested a change only for the sake of improving code readability, or was there a functional aspect as well?

**Comment by igas** [comment]
@anag004 there are no functional differences, however, it's important for at least the following reasons:

- it keeps diff clear in the history;
- for consistency with the rest of codebase;
- and readability.

**Comment by anag004** [comment]
This PR has been completed from my end for quite some time. Will it be merged?

File: Formula/frotz.rb [hunk 4]
```diff
@@ -13,6 +13,11 @@ class Frotz < Formula
     sha256 "887efd67a050038ffb15df8c713eaa40fcbfcc69d36f58373981dba3ccae422b" => :yosemite
   end
 
+  resource "testdata" do
+    url "https://github.com/anag004/FrotzTestFile/blob/master/test.z5?raw=true"
```
Comment (javian) [inline Formula/frotz.rb]: How about using file from the source repo ? https://gitlab.com/DavidGriffith/frotz/tree/master/src/test 
I think that would be more maintainable in the long term.

**Review by javian — CHANGES_REQUESTED** [review]

**Comment by javian** [comment]
I just wanted to note that the github repo of this software has been retired and moved to Gitlab. No new releases has been made but for the long-term this should be changed.

File: Formula/frotz.rb (hunk 4 above)
Comment (anag004) [reply Formula/frotz.rb]: @javian I did consider that. However, those files are designed to debug implementations of Z-machine interpreters and not test an installation. Consequently, these files require user interaction to run which I could not emulate through the `.rb` file.

**Comment by SMillerDev** [comment]
@brewtestbot test this please!

**Comment by javian** [comment]
@BrewTestBot test this please

**Comment by anag004** [comment]
Some test seems to have failed. However, clicking on the details link takes me to a page with a 404 error. Why is this?

**Pull request closed without merging**